### PR TITLE
refactor(service-portal): move delegation + logout buttons

### PR DESCRIPTION
**Pull request by disaerna**
# Service Portal - Sidebar & User Menu changes

## What
* Move delegation button to sidebar
* Add logout button to bottom of sidebar
* Add logout button to bottom of mobile menu

## Why
* Users were having trouble finding delegation button in user testing.

## Screenshots

Sidebar: 
Screenshot 2022-06-07 at 10 47 15

Mobile menu:
Screenshot 2022-06-07 at 10 47 54

User menu:
Screenshot 2022-06-09 at 11 27 32

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/settings/access-control/src/index.ts [hunk 1]
```diff
@@ -12,11 +12,16 @@ export const accessControlModule: ServicePortalModule = {
   name: 'Aðgangsstýring',
   widgets: () => [],
   routes: ({ userInfo }) => {
+    const isCompany = userInfo.profile.subjectType === 'legalEntity'
+    const isDelegation = Boolean(userInfo.profile.actor)
```
Comment (osk) [inline libs/service-portal/settings/access-control/src/index.ts:16]: These checks are done more than once and are not excactly the same. Would recommend creating a helper that ensures consistency, eases refactoring, and allows for a centralized place for documenting behavior (i.e. it's not obvious why `isDelegation = Boolean(userInfo.profile.actor)`)

**Review by osk — APPROVED** [review]

File: libs/service-portal/settings/access-control/src/index.ts (hunk 1 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/index.ts:16]: Agree 🚀 Will refactor in another PR

**Comment by raRaRa** [comment]
Looks good! There's definitely room for refactoring the `NavItem` & `LogOutItem` components since they share so much code. There was another thing I noticed which can be improved as well, which is to remove the `onMouseOver` & `onMouseLeave` events to style the hover effect. Instead we should add one style to the parent `Box` that has a style called `styles.navItem`, that handles the hover effect with a selector, and a globalStyle to target the svg.

Here's the code for the styles:

```
export const navItem = style({
  ':hover': {
    color: theme.color.blue400,
    backgroundColor: theme.color.blue100,
  },
})

globalStyle(`${navItem}:hover svg`, {
  color: theme.color.blue400,
})
```

Using `globalStyle` is great when you need to target a children, e.g. the svg in this case.


The Box component would end up with these styles:

```
      className={[
        styles.navItem,
        styles.navItemActive[navItemActive],
      ]}
```

We could just create a refactoring task for later, but if you feel like it then I challenge you to try the style improvement :)

**Review by raRaRa — APPROVED** [review]
Looks good! There's definitely room for refactoring the `NavItem` & `LogOutItem` components since they share so much code. There was another thing I noticed which can be improved as well, which is to remove the `onMouseOver` & `onMouseLeave` events to style the hover effect. Instead we should add one style to the parent `Box` that has a style called `styles.navItem`, that handles the hover effect with a selector, and a globalStyle to target the svg.

Here's the code for the styles:

```
export const navItem = style({
  ':hover': {
    color: theme.color.blue400,
    backgroundColor: theme.color.blue100,
  },
})

globalStyle(`${navItem}:hover svg`, {
  color: theme.color.blue400,
})
```

Using `globalStyle` is great when you need to target a children, e.g. the svg in this case.


The Box component would end up with these styles:

```
      className={[
        styles.navItem,
        styles.navItemActive[navItemActive],
      ]}
```

We could just create a refactoring task for later, but if you feel like it then I challenge you to try the style improvement :)

**Pull request merged**